eth/eth_test/common_test.go

```diff
@@ -0,0 +1,213 @@
+	auth          *bind.TransactOpts
+	shares        [][]byte
+	execClient    *executionclient.ExecutionClient
+	rpcServer     *rpc.Server
+	httpSrv       *httptest.Server
+	validatorCtrl *mocks.MockController
+	mockCtrl      *gomock.Controller
+}
+
+func (e *TestEnv) shutdown() {
+	if e.mockCtrl != nil {
+		e.mockCtrl.Finish()
+	}
+
+	if e.httpSrv != nil {
+		e.httpSrv.Close()
+	}
+
+	if e.execClient != nil {
+		// Always returns nil error
+		_ = e.execClient.Close()
+	}
+}
+
+func (e *TestEnv) setup(
+	t *testing.T,
+	ctx context.Context,
+	testAddresses []*ethcommon.Address,
+	validatorsCount uint64,
+	operatorsCount uint64,
+) error {
+	logger := zaptest.NewLogger(t)
+
+	// Create operators RSA keys
+	ops, err := createOperators(operatorsCount, 0)
+	if err != nil {
+		return err
+	}
+
+	validators := make([]*testValidatorData, validatorsCount)
+	shares := make([][]byte, validatorsCount)
+
+	// Create validators, BLS keys, shares
+	for i := 0; i < int(validatorsCount); i++ {
+		validators[i], err = createNewValidator(ops)
+		if err != nil {
+			return err
+		}
+
+		shares[i], err = generateSharesData(validators[i], ops, testAddrAlice, i)
+		if err != nil {
+			return err
+		}
+	}
+
+	eh, validatorCtrl, mockCtrl, nodeStorage, err := setupEventHandler(t, ctx, logger, ops[0], &testAddrAlice, true)
+	e.mockCtrl = mockCtrl
+	e.nodeStorage = nodeStorage
+
+	if err != nil {
+		return err
+	}
+	if validatorCtrl == nil {
+		return fmt.Errorf("error: validatorCtrl is empty")
+	}
+
+	// Adding testAddresses to the genesis block mostly to specify some balances for them
+	sim := simTestBackend(testAddresses)
+
+	// Create JSON-RPC handler
+	rpcServer, err := sim.Node.RPCHandler()
+	e.rpcServer = rpcServer
+	if err != nil {
+		return fmt.Errorf("creatingt rpc server: %w", err)
+	}
+	// Expose handler on a test server with ws open
+	httpSrv := httptest.NewServer(rpcServer.WebsocketHandler([]string{"*"}))
+	e.httpSrv = httpSrv
+
+	addr := "ws:" + strings.TrimPrefix(httpSrv.URL, "http:")
+
+	parsed, err := abi.JSON(strings.NewReader(simcontract.SimcontractMetaData.ABI))
+	if err != nil {
+		return fmt.Errorf("parsing contract abi: %w", err)
+	}
+
+	auth, err := bind.NewKeyedTransactorWithChainID(testKeyAlice, big.NewInt(1337))
+	if err != nil {
+		return err
+	}
+
+	contractAddr, _, _, err := bind.DeployContract(auth, parsed, ethcommon.FromHex(simcontract.SimcontractMetaData.Bin), sim)
+	if err != nil {
+		return fmt.Errorf("deploying contract: %w", err)
+	}
+
+	sim.Commit()
+
+	// Check contract code at the simulated blockchain
+	contractCode, err := sim.CodeAt(ctx, contractAddr, nil)
+	if err != nil {
+		return fmt.Errorf("getting contract code: %w", err)
+	}
+	if len(contractCode) == 0 {
+		return fmt.Errorf("error: contractCode is empty")
+	}
+
+	// Create a client and connect to the simulator
+	e.execClient, err = executionclient.New(
+		ctx,
+		addr,
+		contractAddr,
+		executionclient.WithLogger(logger),
+		executionclient.WithFollowDistance(0),
```

Comment:
This means follow distance is never tested, even though it's always used in practice (value is 8 in production). Can we please try increasing this to 1 or 2 and adjust the tests accordingly?

I imagine that the major difference is that the tests would have to produce an additional `followDistance` number of empty blocks after a block with transactions to make the execution client sync the transactions. Probably we can just do `for i := 0; i < followDistance; i++ { Commit() }` each time after producing a block with a tx that we want to test. This can probably be automated in a special `CloseFollowDistance` method under `TestEnv`, so the test may commit its transactions in however many blocks it wants and then we may call this method to make sure we sync them.

Additionally, when confirming sync height, instead of comparing `lastProcessedBlockNumber` to `currentBlockNumber`, you may have to compare to `currentBlockNumber - followDistance`.

Reply:
Added this. Thanks. Good point

Reply:
> Added this. Thanks. Good point

Thanks!